API/models/rbac.go

```diff
@@ -0,0 +1,122 @@
+package models
+
+import (
+	u "p3/utils"
+	"strings"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+// Roles
+const (
+	Manager = "manager"
+	User    = "user"
+	Viewer  = "viewer"
+)
+
+// Actions
+const (
+	READ = iota
+	WRITE
+	READONLYNAME
+)
```

Comment:
These should have a proper type rather than a comment. Like
```go
type Role string
type Permission string
```
this way the CheckUserPermissions function would have a clearer signature